server/controllers/users.js

```diff
@@ -92,11 +95,10 @@ export const updateToken = async (req, res) => {
 };
 
 /**
- * Verify the 2FA code the user has entered when
- * logging in and send back another JWT.
+ * Verify the 2FA code or recovery code the user has entered when logging in and send back another JWT.
  */
 export const twoFactorAuthAndUpdateToken = async (req, res, next) => {
-  const { twoFactorAuthenticatorCode } = req.body;
+  const { twoFactorAuthenticatorCode, recoveryCode } = req.body;
```

Comment:
There is one parameter prefixed with `twoFactorAuthenticator` and the other not. You should try to be consistent in your variable naming.